pkg/util/pdag/pdag.go

```diff
@@ -18,24 +18,57 @@ package pdag
 
 import (
 	"context"
+	"errors"
 	"iter"
 	"sync"
-	"sync/atomic"
 
 	"golang.org/x/sync/errgroup"
 )
 
-type DAG[T any] struct{ nodes []node[T] }
+func New[T any]() *DAG[T] {
+	return &DAG[T]{m: *sync.NewCond(&sync.Mutex{})}
+}
+
+type DAG[T any] struct {
+	m     sync.Cond
+	nodes []node[T]
+}
 
 type node[T any] struct {
 	v T
-	// the list edge destinations for edges originating from this node.
-	to []int
+
+	// the non-transitive list of nodes that must be processed before this node.
```

Comment:
`non-transitive list of nodes` 

Does this mean the direct prerequisites, ie if I have `A -> B -> C`, for the node `C` this would be `[B]`?